File: leet_LIST_234_is_palindrome___brute.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head)
    {
        Stack<Integer>  s;
        ListNode        node;

        s = new Stack<Integer>();
        node = head;
        while (node != null)
        {
            s.push(node.val);
            node = node.next;
        }
        node = head;
        while (node != null && !s.empty())
        {
            if (s.pop() != node.val)
                return false ;
            node = node.next;
        }
        return true;
    }
}

File: leet_LIST_234_is_palindrome___brute.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        d = []
        p = head
        while p:
            d.append(p.val)
            p = p.next
        p = head
        while p:
            if p.val != d.pop():
                return False
            p = p.next
        return True
